### PR TITLE
LPS-144708,LPS-144709

### PR DESCRIPTION
**Pull request by rebsilva**
Related Issues:
- https://issues.liferay.com/browse/LPS-144708
- https://issues.liferay.com/browse/LPS-144709

Hello Reviewer! 🌸

This PR has new features related to the story:
**As a form admin, I want to be able to request datetime from my users** ([LPS-143050](https://issues.liferay.com/browse/LPS-143050))

Summary: Updates the date picker dependency, creates the `use12Hours` variable and formats the value of input hidden according to the field type.
More details can be found [here](https://liferay.atlassian.net/wiki/spaces/ENGFORMS/pages/1958839342/LPS-143050+As+a+form+admin+I+want+to+be+able+to+request+datetime+from+my+users) on our Confluence page.

If you have any questions feel free to ask, thank you for your time!

### CODE REVIEW

**Comment by liferay-forms** [comment]
Hi, you don’t need to run the CI commands anymore, I’ll do it for you and let you know if any issue happens.

**Comment by liferay-forms** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @rebsilva, can you please check if these failures are related to your changes? Thanks

**Comment by rebsilva** [comment]
ci:reevaluate:1277401_1296

**Comment by rebsilva** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js [hunk 1]
```diff
@@ -368,37 +375,52 @@ const Main = ({
...
+}) => {
+	let use12Hours;
+
+	const isDateTime = type === 'date_time';
+
+	if (isDateTime) {
+		const momentLocale = moment().locale(locale);
+		const time = momentLocale.localeData().longDateFormat('LT');
+		use12Hours = time.endWith('A');
+	}
```
Comment (alinedoleron) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js:390]: Hi @rebsilva, sorry for the typo on the documentation. Itś missing an 's' on endWith method. Please, change `time.endWith('A')`  by `time.endsWith('A');`

**Review by alinedoleron — CHANGES_REQUESTED** [review]
Great job, @rebsilva. There's just one comment which was a problem on the refinement documentation.

**Comment by rebsilva** [comment]
I will close this PR because I opened a [new one](https://github.com/liferay-forms/liferay-portal/pull/1367) about this story, thanks!

**Pull request closed without merging**